Week6/news/src/app/news.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from "@angular/core";
+import { newsApiEndPoint } from "./Config/Endpoint";
+import { HttpClient } from "@angular/common/http";
+import { Observable } from "rxjs";
+
+@Injectable({
+  providedIn: "root"
+})
+export class NewsService {
+  public sourceName = "bbc-news";
+  public articles = [];
+  constructor(private http: HttpClient) {
+    this.getAuthorsList(this.sourceName).then(data => (this.articles = data));
```

Comment:
constructor should not contain complex/blocking operations. You just have to call the getAuthorsList method from component in onInit.